lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java

```diff
@@ -346,6 +352,7 @@ private void readAffixFile(InputStream affixStream, CharsetDecoder decoder, Flag
       if (line.isEmpty()) continue;
 
       String firstWord = line.split("\\s")[0];
+      // TODO: convert to a switch?
```

Comment:
I thought about that. Maybe switch expression, when the language level allows that. Switch with break statements would be too verbose for my taste

Reply:
Ok. My taste tells me it'd be cleaner than that multi-level-if, especially that not all statements are identical there (one compares to the entire line if I believe). It may be worth considering the switch for performance reasons too but don't know if you'd see the difference.

Reply:
A bit cleaner, yes, but verbosity and the risk of forgetting a `break` outweigh that for me.
I also considered creating a map from first word into parsing lambdas, but decided it'd be quite verbose as well.

Performance should be negligible here. Last time I checked, it was dominated by writing/sorting/reading dic entries and building FSTs.

Reply:
ok.